homeassistant/components/keymitt_ble/config_flow.py

```diff
@@ -139,7 +139,11 @@ async def async_step_link(
             return self.async_show_form(step_id="link")
 
         if not await self._client.is_connected():
-            errors["base"] = "linking"
+            await self._client.connect(init=False)
+            if not await self._client.is_connected():
+                errors["base"] = "linking"
+            else:
+                await self._client.disconnect()
```

Comment:
```suggestion
        if not await self._client.is_connected():
            errors["base"] = "linking"
```
Maybe this makes more sense, so it doesn't look as confusing